src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs

```diff
@@ -122,11 +123,21 @@ public async Task<Solution> CreateSolutionAsync(Checksum newSolutionChecksum, Ca
                         var newVersions = await _assetProvider.GetAssetAsync<SourceGeneratorExecutionVersionMap>(
                             AssetPathKind.SolutionSourceGeneratorExecutionVersionMap, newSolutionCompilationChecksums.SourceGeneratorExecutionVersionMap, cancellationToken).ConfigureAwait(false);
 
-                        // The execution version map will be for the entire solution on the host side.  However, we may
-                        // only be syncing over a partial cone.  In that case, filter down the version map we apply to
-                        // the local solution to only be for that cone as well.
-                        newVersions = FilterToProjectCone(newVersions, newSolutionChecksums.ProjectCone);
-                        solution = solution.WithSourceGeneratorExecutionVersions(newVersions, cancellationToken);
+#if DEBUG
+                        var projectCone = newSolutionChecksums.ProjectCone;
+                        if (projectCone != null)
+                        {
+                            Debug.Assert(projectCone.ProjectIds.Count == newVersions.Map.Count);
+                            Debug.Assert(projectCone.ProjectIds.All(id => newVersions.Map.ContainsKey(id)));
```

Comment:
Do we not have a SetEquals helper to call? This works but would be shorter/clearer.

Reply:
def can clean that up in followup.